### PR TITLE
feat(composition): support @cost and @listSize

### PR DESCRIPTION
**Pull request by Aenimus**
## Summary by CodeRabbit

* **New Features**
  * Added @cost and @listSize directives to declare field/argument weights and per-field list-size metadata; these are carried through composition into router configurations.
* **Documentation**
  * New Composition Architecture & Onboarding guide and README additions explaining the composition flow and directive handling.
* **Tests**
  * Comprehensive test suites for @cost and @listSize covering normalization, federation behavior, validation, and edge cases.
* **Chores**
  * Composition and config tooling now include optional cost/list-size data in composed outputs.

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: composition/src/v1/federation/federation-factory.ts [hunk 1]
```diff
@@ -278,8 +282,10 @@ export class FederationFactory {
   parentTagDataByTypeName = new Map<TypeName, ParentTagData>();
   persistedDirectiveDefinitionByDirectiveName = new Map<DirectiveName, DirectiveDefinitionNode>([
```
Comment (Aenimus) [inline composition/src/v1/federation/federation-factory.ts:279]: These directives are not persisted in the client schema, so they don't need to go in here.

File: composition/tests/v1/directives/listSize.test.ts [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+            type User {
+              id: ID!
+            }
+          `,
+        ),
+      );
+    });
+
+    test('that multiple @listSize directives on different fields are preserved in federated schema', () => {
+      const { federatedGraphSchema } = federateSubgraphsSuccess(
```
Comment (Aenimus) [inline composition/tests/v1/directives/listSize.test.ts]: We should also check the client schema (which should not contain the directives)

**Comment by ysmolski** [comment]
Quick question. Not sure if it can be reviewed. Do we forbid costs on fields of interfaces?

File: composition/src/v1/federation/federation-factory.ts (hunk 1 above)
Comment (StarpTech) [reply composition/src/v1/federation/federation-factory.ts:279]: Done

**Comment by StarpTech** [comment]
> Do we forbid costs on fields of interfaces?

@ysmolski please check the spec and https://www.apollographql.com/docs/graphos/schema-design/federated-schemas/reference/directives#cost

File: composition/tests/v1/directives/listSize.test.ts (hunk 2 above)
Comment (StarpTech) [reply composition/tests/v1/directives/listSize.test.ts]: Done

File: composition/tests/v1/directives/cost.test.ts [hunk 3]
```diff
@@ -0,0 +1,626 @@
...
+            type Query {
+              fieldA: String! @cost(weight: "10")
+              fieldB: String! @cost(weight: "20")
+            }
+          `,
+        ),
+      );
+    });
+
+    test('that @cost from multiple subgraphs on the same entity field is deduplicated to first instance', () => {
```
Comment (StarpTech) [inline composition/tests/v1/directives/cost.test.ts]: @Aenimus Not sure about this behaviour

**Comment by ysmolski** [comment]
@StarpTech 
This is the structure per data source where engine store everything related to cost directives: https://github.com/wundergraph/graphql-go-tools/blob/96bc286c1e4bb9a28f076c4ca486e679fc5d318e/v2/pkg/engine/plan/static_cost.go#L120

File: composition/tests/v1/directives/cost.test.ts (hunk 3 above)
Comment (ysmolski) [reply composition/tests/v1/directives/cost.test.ts]: This test is not needed. Cost on each subgraph should be isolated and not deduplicated. I will remove the test

File: composition/src/v1/normalization/normalization-factory.ts [hunk 4]
```diff
@@ -2314,6 +2369,167 @@ export class NormalizationFactory {
...
+              listSizeSlicingArgumentNotIntErrorMessage(directiveCoords, slicingArgName, printTypeNode(argData.type)),
+            );
+            continue;
+          }
+          if (argData.namedTypeName !== INT_SCALAR) {
+            errorMessages.push(
+              listSizeSlicingArgumentNotIntErrorMessage(directiveCoords, slicingArgName, printTypeNode(argData.type)),
+            );
+          }
+        }
```
[the remark this answers is not in the document]
Comment (ysmolski) [reply composition/src/v1/normalization/normalization-factory.ts]: check that it was fixed
